src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardUserInfoDirectives.scala

```diff
@@ -26,11 +26,28 @@ trait StandardUserInfoDirectives extends UserInfoDirectives with LazyLogging wit
 
   def requireActiveUser(samRequestContext: SamRequestContext): Directive1[SamUser] = requireOidcHeaders.flatMap { oidcHeaders =>
     onSuccess {
-      getUserInfo(directoryDAO, registrationDAO, oidcHeaders, samRequestContext).unsafeToFuture()
+      val requireActiveUserIO = for {
+        user <- getSamUser(directoryDAO, registrationDAO, oidcHeaders, samRequestContext)
+        tosStatus <- tosService.getTosStatus(user.id, samRequestContext)
```

Comment:
Note that this is an extra db call per api call to sam but it should be a few millis. This can be wedged into an existing call but let's not optimize prematurely.